### PR TITLE
Host of changes to bring Docker-Provider project onto github

### PR DESCRIPTION
**Pull request by jeffaco**
@Microsoft/ostc-devs @Microsoft/omsagent-devs 

There are a number of commits here, each logical on their own. Or just look over all of the changes.

Sorry for spacing and line ending changes. I wanted files in consistent form on GitHub.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# git ignore file for the Docker-Provider project
+
+/intermediate/
+/target/
+build/config.mak
+
+# Unit test files
+
+/test/code/providers/TestScriptPath.h
```
Comment (niroyb) [inline .gitignore]: Why are these test files ignored?

File: build/Makefile [hunk 2]
```diff
@@ -14,6 +14,13 @@ ifndef ENABLE_DEBUG
 $(error "ENABLE_DEBUG is not set.  Please re-run configure")
 endif
 
+# Include the version file
+include ../../docker.version
+
+ifndef CONTAINER_BUILDVERSION_STATUS
+$(error "Is Makefile.version missing?  Please re-run configure")
```
Comment (niroyb) [inline build/Makefile]: If `CONTAINER_BUILDVERSION_STATUS` is defined in `docker.version` then the error message should be fixed.

File: build/Makefile [hunk 3]
```diff
@@ -255,7 +259,7 @@ $(PROVIDER_LIBRARY): CFLAGS += $(PROVIDER_COMPILE_FLAGS)
 $(PROVIDER_LIBRARY): CXXFLAGS += $(PROVIDER_COMPILE_FLAGS)
 $(PROVIDER_LIBRARY): $(STATIC_PROVIDERLIB_OBJFILES) $(STATIC_PROVIDERLIB_LOGPOLICY) $(INCLUDE_DEFINES) $(PROVIDER_HEADERS)
 	$(MKPATH) $(INTERMEDIATE_DIR)
-	$(MKPATH) $(TARGET_DIR)
+	$(MKPATH) $(INTERMEDIATE_DIR)
```
Comment (niroyb) [inline build/Makefile]: Duplicate line with the previous one

Comment (niroyb) [inline build/Makefile]: Duplicate line with the previous one

File: build/Makefile [hunk 4]
```diff
@@ -286,12 +290,12 @@ STATIC_PROVIDER_PAL_UNITFILES = \
...
 
-$(TARGET_DIR)/testrunner: INCLUDES += $(PROVIDER_TEST_INCLUDE_FLAGS) -I$(PAL_INCLUDE_DIR) -I$(PAL_TESTUTILS_DIR) -I$(PROVIDER_DIR) -I$(SOURCE_DIR)
-$(TARGET_DIR)/testrunner: CFLAGS += $(PROVIDER_COMPILE_FLAGS)
-$(TARGET_DIR)/testrunner: CXXFLAGS += $(PROVIDER_COMPILE_FLAGS)
-$(TARGET_DIR)/testrunner : $(STATIC_PROVIDER_TEST_OBJFILES) $(STATIC_PROVIDERLIB_OBJFILES) $(INCLUDE_DEFINES) $(PROVIDER_HEADERS)
+$(INTERMEDIATE_DIR)/testrunner: INCLUDES += $(PROVIDER_TEST_INCLUDE_FLAGS) -I$(PAL_INCLUDE_DIR) -I$(PAL_TESTUTILS_DIR) -I$(PROVIDER_DIR) -I$(SOURCE_DIR)
+$(INTERMEDIATE_DIR)/testrunner: CFLAGS += $(PROVIDER_COMPILE_FLAGS)
+$(INTERMEDIATE_DIR)/testrunner: CXXFLAGS += $(PROVIDER_COMPILE_FLAGS)
+$(INTERMEDIATE_DIR)/testrunner : $(STATIC_PROVIDER_TEST_OBJFILES) $(STATIC_PROVIDERLIB_OBJFILES) $(INCLUDE_DEFINES) $(PROVIDER_HEADERS)
+	$(MKPATH) $(INTERMEDIATE_DIR)
```
Comment (niroyb) [inline build/Makefile]: Duplicate with the following line.

File: build/Makefile [hunk 5]
```diff
@@ -361,20 +365,24 @@ ifeq ($(ULINUX),1)
...
+	sed -re 's/.rpm$$|.deb$$//' $(INTERMEDIATE_DIR)/package_filename > $(INTERMEDIATE_DIR)/package_file.tmp; mv $(INTERMEDIATE_DIR)/package_file.tmp $(INTERMEDIATE_DIR)/package_filename
 
 	# Build the tar file containing both .rpm and .deb packages
-	cd $(TARGET_DIR); tar cvf $(OUTPUT_PACKAGE_PREFIX).tar $(OUTPUT_PACKAGE_PREFIX).rpm $(OUTPUT_PACKAGE_PREFIX).deb
+	cd $(INTERMEDIATE_DIR); tar cvf $(OUTPUT_PACKAGE_PREFIX).tar $(OUTPUT_PACKAGE_PREFIX).rpm $(OUTPUT_PACKAGE_PREFIX).deb
+
+	../installer/bundle/create_bundle.sh $(PF)_$(PF_DISTRO) $(INTERMEDIATE_DIR) $(OUTPUT_PACKAGE_PREFIX)
+	# Copy the shell bundle to the target directory
+	mkdir -p $(TARGET_DIR)
+	cd $(INTERMEDIATE_DIR); cp `cat $(INTERMEDIATE_DIR)/package_filename`.sh $(TARGET_DIR)
```
Comment (niroyb) [inline build/Makefile]: The `cd` looks unnecessary.

**Comment by niroyb** [comment]
Signed off

File: .gitignore (hunk 1 above)
Comment (jeffaco) [inline .gitignore]: This file is created by the build process. I didn't get into details with Yuyang about it, I didn't much care. It's created during the build process, so I put it in .gitignore.

File: build/Makefile (hunk 2 above)
Comment (jeffaco) [inline build/Makefile]: Yeah, that's true. This is Makefile.version in other stuff. Here, I just decided to make this fixed (no need to have a temporary one, we only support building with superproject).

I'll fix the comment.

File: build/Makefile (hunk 3 above)
Comment (jeffaco) [inline build/Makefile]: Global search replace, missed that. Fixed.

File: build/Makefile (hunk 5 above)
Comment (jeffaco) [inline build/Makefile]: It's not. The package_filename file doesnt contain a path prefix so, without adding that, this will fail without the `cd`.

**Pull request merged**